src/tasks/component-installers/dex.ts

```diff
@@ -0,0 +1,326 @@
+                  await this.kube.createClusterRoleBindingRoleFromFile(yamlFilePath)
+                  task.title = `${task.title}...[OK]`
+                }
+              },
+            },
+            {
+              title: 'Create Dex service',
+              task: async (_ctx: any, task: any) => {
+                if (await this.kube.isServiceExists(this.dexName, this.namespaceName)) {
+                  task.title = `${task.title}...[Exists]`
+                } else {
+                  const yamlFilePath = this.getDexResourceFilePath('service.yaml')
+                  await this.kube.createServiceFromFile(yamlFilePath, this.namespaceName)
+                  task.title = `${task.title}...[OK]`
+                }
+              },
+            },
+            {
+              title: 'Create Dex ingress',
+              task: async (_ctx: any, task: any) => {
+                if (await this.kube.isIngressExist(this.dexName, this.namespaceName)) {
+                  task.title = `${task.title}...[Exists]`
+                } else {
+                  const yamlFilePath = this.getDexResourceFilePath('ingress.yaml')
+                  let yamlContent = fs.readFileSync(yamlFilePath).toString()
+                  yamlContent = yamlContent.replace(new RegExp(TemplatePlaceholders.DOMAIN, 'g'), this.flags.domain)
+
+                  const ingress = yaml.load(yamlContent) as V1Ingress
+                  await this.kube.createIngressFromObj(ingress, this.namespaceName)
+
+                  task.title = `${task.title}...[OK]`
+                }
+              },
+            },
+            {
+              title: 'Generate Dex username and password',
+              task: async (ctx: any, task: any) => {
+                const dexConfigMap = await this.kube.getConfigMap(this.dexName, this.namespaceName)
+                if (dexConfigMap && dexConfigMap.data) {
+                  task.title = `${task.title}...[Exists]`
+                } else {
+                  const dexPassword = generatePassword(12)
+
+                  const salt = bcrypt.genSaltSync(10)
+                  const dexPasswordHash = bcrypt.hashSync(dexPassword, salt)
+
+                  ctx[DexContextKeys.DEX_USERNAME] = 'admin'
+                  ctx[DexContextKeys.DEX_PASSWORD] = dexPassword
+                  ctx[DexContextKeys.DEX_PASSWORD_HASH] = dexPasswordHash
+
+                  task.title = `${task.title}...[OK: ${ctx[DexContextKeys.DEX_USERNAME]}:${ctx[DexContextKeys.DEX_PASSWORD]}]`
+                }
+              },
+            },
+            {
+              title: 'Create Dex configmap',
+              task: async (ctx: any, task: any) => {
+                const dexConfigMap = await this.kube.getConfigMap(this.dexName, this.namespaceName)
+                if (dexConfigMap && dexConfigMap.data) {
+                  // read client secret
+                  const config = yaml.load(dexConfigMap.data['config.yaml']) as any
```

Comment:
Are you sure, that `yaml.load` tolerate `undefined` as an argument?

Reply:
Agree. Will fix.